### PR TITLE
Update deps, add ruff

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ jobs:
     name: Test
     strategy:
       matrix:
-        python-version: ['3.6.x', '3.7.x', '3.8.x']
+        python-version: ['3.7.x', '3.8.x']
```
Comment (rowanseymour) [inline .github/workflows/ci.yml]: Make this 3.9.x, 3.10.x, 3.11.x

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -24,9 +24,20 @@ jobs:
     - name: Initialize environment
       run: poetry install
 
+    - name: Run black
+      run: |
```
Comment (rowanseymour) [inline .github/workflows/ci.yml]: don't need to use `|` if you're just running one line

File: pyproject.toml [hunk 3]
```diff
@@ -20,16 +20,36 @@ packages = [
 repository = "http://github.com/rapidpro/rapidpro-python"
 
 [tool.poetry.dependencies]
-python = "^3.6"
+python = "^3.7"
```
Comment (rowanseymour) [inline pyproject.toml]: can make this 3.9

File: pyproject.toml [hunk 4]
```diff
@@ -20,16 +20,36 @@ packages = [
...
+
+[tool.black]
+line-length = 119
+
+[tool.ruff]
+line-length = 120
+select = ["E", "F", "W"]
+ignore = ["E501", "F405"]
+fix = true
+exclude = ["./.tox/*", "./env/*", "./.venv/*", "./build/*", "./dist/*", "./docs/*"]
```
Comment (rowanseymour) [inline pyproject.toml]: do we need this?

File: temba_client/v2/__init__.py [hunk 5]
```diff
@@ -131,7 +131,16 @@ def get_classifiers(self, uuid=None):
         params = self._build_params(uuid=uuid)
         return self._get_query("classifiers", params, Classifier)
 
-    def get_contacts(self, uuid=None, urn=None, group=None, deleted=None, before=None, after=None, reverse=None):
```
Comment (rowanseymour) [inline temba_client/v2/__init__.py:134]: did you maybe accidentally format this as line-length 80? these lines don't look so long

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Comment by norkans7** [comment]
nose not working with python 3.10, I will check a solution for that

File: temba_client/v2/__init__.py (hunk 5 above)
Comment (norkans7) [reply temba_client/v2/__init__.py:134]: updated by re-applying the latest config

**Pull request merged**